File: Programmers/노수일/Q42586.py
def solution(progresses, speeds):
    answer = []
    next_seq = 0

    while next_seq < len(progresses):
        numbers = 0
        for i in range(next_seq, len(progresses)):
            progresses[i] = progresses[i] + speeds[i]

        for i in range(next_seq, len(progresses)):
            if progresses[i] >= 100:
                next_seq = i + 1
                numbers = numbers + 1
            else:
                break

        if numbers != 0:
            answer.append(numbers)

    return answer

File: Programmers/정권호/Q42586.java
package programers.lv2._0526;

import java.util.ArrayList;
import java.util.List;

public class Q42586 {
    class Solution {
        public int[] solution(int[] progresses, int[] speeds) {
            int[] answer = {};
            int[] days = new int[progresses.length];

            for (int i = 0; i < progresses.length; i++) {
                int day = (100 - progresses[i]) / speeds[i];
                if ((100 - progresses[i]) % speeds[i] != 0) {
                    day++;
                }
                days[i] = day;
            }

            List<Integer> answers = new ArrayList<>();
            int distribution = days[0];
            int cnt = 1;

            for (int i = 1; i < days.length; i++) {
                if (distribution < days[i]) {
                    answers.add(cnt);
                    cnt = 1;
                    distribution = days[i];
                    continue;
                }
                if (i == days.length - 1) {
                    answers.add(cnt);
                }
                cnt++;
            }

            answer = new int[answers.size()];

            for (int i = 0; i < answers.size(); i++) {
                answer[i] = answers.get(i);
            }

            return answer;
        }
    }
}
